src/App.tsx

```diff
@@ -8,7 +7,73 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+import { TodoFilterValues } from './types/filterEnum';
+
 export const App: React.FC = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [choosedTodo, setChoosedTodo] = useState<Todo | null>(null);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [
+    todoFilter,
+    setTodoFilter,
+  ] = useState<TodoFilterValues>(TodoFilterValues.all);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
```

Comment:
Let’s use `async await` all over the code. It gives you a simpler way to work with asynchronous promise-based code